Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -55,31 +57,98 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
 
             internalBoundaries = internalBoundaries ?? new List<ICurve>();
 
-            foreach (ICurve crv in internalBoundaries)
+            //----------------Closed-Internal-Boundaries--------------------//
+            int count = internalBoundaries.Count;
+
+            internalBoundaries = internalBoundaries.Where(x => x.IIsClosed()).ToList();
+
+            if (internalBoundaries.Count != count)
             {
-                if (!crv.IIsPlanar())
-                {
-                    Reflection.Compute.RecordError("At least one of the internal edge curves is not planar");
-                    return null;
-                }
+                Reflection.Compute.RecordWarning("At least one of the internal boundaries is not closed and has been ignored on creation of the planar surface.");
+            }
 
-                if (!crv.IIsClosed())
-                {
-                    Reflection.Compute.RecordError("At least one of the internal edge curves is not closed");
-                    return null;
-                }
+            //---------------Coplanar-Internal-Boundaries-------------------//
+            count = internalBoundaries.Count;
+
+            Plane p = externalBoundary.IFitPlane();
+            internalBoundaries = internalBoundaries.Where(x => x.IIsInPlane(p)).ToList();
```

Comment:
`IsInPlane` is not implemented for `Ellipse` and `NurbsCurve` - sounds like a 5 min job, possibly could be added in this PR?

Reply:
Just assumed that it would be, but yes ofc